funcx_endpoint/funcx_endpoint/endpoint/utils/config.py

```diff
@@ -82,16 +83,35 @@ def __init__(
         log_dir=None,
         stdout="./endpoint.log",
         stderr="./endpoint.log",
+        **kwargs,
     ):
+        for deprecated_name in ("results_ws_uri", "warn_about_url_mismatch"):
+            if deprecated_name in kwargs:
+                caller_filename = inspect.stack()[1].filename
+                msg = (
+                    f"The '{deprecated_name}' argument is deprecated.  It will"
+                    f" be removed in a future release.  Found in: {caller_filename}"
+                )
+                warnings.warn(msg)
+                del kwargs[deprecated_name]
+
+        for unknown_arg in kwargs:
+            # Calculated multiple times, but only in errant case -- nominally
+            # fixed and then "not an issue."
+            caller_filename = inspect.stack()[1].filename
+            cls_name = self.__class__.__name__
+            msg = (
+                f"Unknown argument to {cls_name} ignored: {unknown_arg}\n"
+                f"\n  Specified in: {caller_filename}"
```

Comment:
Why is there two newlines here?  Is that standard for warnings?

Reply:
Addressed, and also rebased with Chris' safety fix from main.  Let's see how this goes ...